### ISSUE TITLE
Add support for temporarily generated access tokens

### ISSUE THREAD

**Issue by rjerrems** [terraform-google-modules/terraform-google-project-factory#243]
It would be great to support temporarily generated access tokens with this module. Right now, if temporary tokens are used with Terraform provider configuration the module mostly works - except for cases where gcloud commands are used or when the preconditions python script runs as they both assume they are using a static credentials file. An example of my provider setup looks like this:
```
locals {
  tf_sa = "${var.terraform_service_account}"
}

provider "google" {
  alias = "impersonate"

  scopes = [
    "https://www.googleapis.com/auth/cloud-platform",
    "https://www.googleapis.com/auth/userinfo.email",
  ]
}

data "google_service_account_access_token" "default" {
  provider               = "google.impersonate"
  target_service_account = "${local.tf_sa}"
  scopes                 = ["userinfo-email", "cloud-platform"]
  lifetime               = "600s"
}

provider "google" {
  access_token = "${data.google_service_account_access_token.default.access_token}"
  version      = "~> 2.1"
}

provider "google-beta" {
  access_token = "${data.google_service_account_access_token.default.access_token}"
  version      = "~> 2.1"
}
```

If using a setup as mentioned as above, a possible work around in the case of using gcloud commands is to use the impersonation flag and pass the service account to impersonate like this:
```
gcloud compute instances list --impersonate-service-account= "${local.tf_sa}"
``` 
which would work in the example case. If an additional optional parameter for a service account were added, a shell wrapper for gcloud commands could check if that exists and use the impersonation variation of the gcloud command.

Although the caveat with this approach is it makes the assumption that there is a underlying account which has a credential source either using a metadata endpoint from within GCP or a local credential that is able to generate a token for that service account - it would not work in the case where you have used a external tool to generate a temporary token.

I would be happy to write up a PR for this if this would useful for the module.

**Comment by mikhail-khodorovskiy** [comment]
+ 1!
Just spent a day trying to figure out why the default service account can't be deleted while impersonation is used.

**Comment by morgante** [comment]
Fixed by #243.
